core/shared/src/main/scala/org/http4s/headers/X-Content-Type-Options.scala

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2013 http4s.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.http4s
+package headers
+
+import cats.data.NonEmptyList
+import cats.parse.Parser
+import org.http4s.internal.parsing.CommonRules
+import org.typelevel.ci.CIString
+import org.typelevel.ci.CIStringSyntax
+
+/** Response header that indicates whether the browser should be allowed to
+  * render the response as a downloadable file.
+  *
+  * [[https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Content-Type-Options X-Content-Type-Options]]
+  */
+sealed abstract case class `X-Content-Type-Options` private (value: CIString)
+
+object `X-Content-Type-Options`
+    extends HeaderCompanion[`X-Content-Type-Options`]("X-Content-Type-Options") {
+
+  private class XContentTypeOptionsImpl(value: CIString) extends `X-Content-Type-Options`(value)
+
+  private val nosniff = new XContentTypeOptionsImpl(ci"nosniff")
```

Comment:
Perhaps, it would make sense to make this field public and explicitly typed, like
```suggestion
  val nosniff: `X-Content-Type-Options` = new XContentTypeOptionsImpl(ci"nosniff")
```
That would provide an ability to get a correct header instance without having the parser involved.